File: C++/C++Homework/C++Homework/gamepiece.cpp
/*
 * This is program implements the gamepiece class
 * it contains a label for the gamepiece and
 * other necessary class member functions
 *
 * Completion time: 1 hour
 *
 * @author Gunnar Hoglund
 * @version 1.0
 */
#include <iostream>
#include "gamepiece.h"

// assign the default string to the
// game pieces label
GamePiece::GamePiece(void)
{
    strcpy(label, "---");
}

// overloaded constructor
GamePiece::GamePiece(char* newLabel)
{
    strcpy(label, newLabel);
}

// getter function for the label attribute
char* GamePiece::getLabel()
{
    return label;
}

// trims or pads the label as needed
char* GamePiece::toString()
{
    // get the length of the string
    int i = 0; // length of the c string
    while( label[i] != '\0')
    {
        i++;
    }
    
    // padding or trimming string as needed
    if (i == 3)
    {
        return label;
    }
    else if (i > 3)
    {
        char label_temp[4];
        
        // storing the first three characterts
        strncpy(label_temp, label, 3);
        label_temp[3] = '\0'; // manually adding the null character
        
        strcpy(label, label_temp);
        
        return label;
    }
    else
    {
        // if i < 3 loop through and add however many
        // spaces to the end of the string
        int diff = 3 - i;
        
        // adding however many spaces needed
        for (int j = 1; j <= diff; j++) {
            label[i] = ' ';
        }
        
        label[3] = '\0'; // manually adding the null character
        
        return label;
    }

}

File: Java/Homework/src/app/GamePiece.java
/**

* This program defines a class for a GamePiece object that stores
* a label for the object. It can also truncate or pad the label as needed.

* Completion time: 0.5 hours

*

* @author Gunnar Hoglund

* @version 1.0

*/
package app;

public class GamePiece {

	// instance variable
	private String label;

	public GamePiece() {
		this.label = "---";
	}

	public GamePiece(String newLabel) {
		this.label = newLabel;
	}

	public String getLabel() {
		return label;
	}

	// constructs a string with 3 characters
	public String toString() {

		String NewString = label;

		if (label.length() == 3) {
			NewString = label;
		} else if (label.length() > 3) {
			// getting the first three letters of the
			// string if its too long
			NewString = label.substring(0, 3);
		} else {

			char space = ' ';
			int numSpaces = 3 - label.length();

			// appending spaces to the string if its
			// too short
			for (int i = 0; i < numSpaces; i++) {
				NewString += space;
			}

		}
		return NewString;
	}
}
